core/src/components/segment-button/segment-button.tsx

```diff
@@ -79,6 +79,10 @@ export class SegmentButton implements ComponentInterface, ButtonInterface {
     };
   }
 
+  componentDidLoad() {
+    this.updateState();
```

Comment:
Why does this work? `this.updateState` is already called in `connectedCallback`, so by the time `componentDidLoad` is called this should already have been run.